static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js

```diff
@@ -0,0 +1,152 @@
+import fastdom from 'fastdom';
+import ophan from 'ophan/ng';
+import userAccount from 'common/modules/navigation/user-account';
+
+const html = qwery('html')[0];
+const menuItems = qwery('.js-close-nav-list');
+const enhanced = {};
+
+const weShouldEnhance = (checkbox: HTMLInputElement): ?boolean =>
+    !enhanced[checkbox.id] && checkbox && !checkbox.checked;
+
+const closeAllOtherPrimaryLists = (targetItem?: HTMLElement): void => {
+    menuItems.forEach(item => {
+        if (item !== targetItem) {
+            item.removeAttribute('open');
+        }
+    });
+};
+
+const removeOrderingFromLists = (): void => {
+    const mainListItems = qwery('.js-navigation-item');
+
+    mainListItems.forEach(item => {
+        const listItem = item;
+
+        listItem.style.order = '';
+    });
+};
+
+const toggleSidebar = (event: Event): void => {
+    // #? hacky type cast
+    const button: HTMLElement = (event.target: any);
+    const mainMenu = document.getElementById('main-menu');
+    const veggieBurgerLink = qwery('.js-change-link')[0];
+
+    const menuIsOpen = (): boolean =>
+        button.getAttribute('aria-expanded') === 'true';
+
+    if (!mainMenu || !veggieBurgerLink) {
+        return;
+    }
+    if (menuIsOpen()) {
+        fastdom.write(() => {
+            button.setAttribute('aria-expanded', 'false');
+            mainMenu.setAttribute('aria-hidden', 'true');
+            veggieBurgerLink.classList.remove(
+                'new-header__nav__menu-button--open'
+            );
+            veggieBurgerLink.setAttribute(
+                'data-link-name',
+                'nav2 : veggie-burger : show'
+            );
+            removeOrderingFromLists();
+
+            // Users should be able to scroll again
+            html.classList.remove('nav-is-open');
+        });
+    } else {
+        fastdom.write(() => {
+            const firstButton = qwery('.js-navigation-button')[0];
+
+            button.setAttribute('aria-expanded', 'true');
+            mainMenu.setAttribute('aria-hidden', 'false');
+            veggieBurgerLink.classList.add(
+                'new-header__nav__menu-button--open'
+            );
+            veggieBurgerLink.setAttribute(
+                'data-link-name',
+                'nav2 : veggie-burger : hide'
+            );
+
+            if (firstButton) {
+                firstButton.focus();
+            }
+            // No targetItem to put in as the parameter. All lists should close.
+            closeAllOtherPrimaryLists();
+            // Prevents scrolling on the body
+            html.classList.add('nav-is-open');
+        });
+    }
+};
+
+const applyEnhancementsTo = (checkbox: HTMLElement): void => {
+    fastdom.read(() => {
+        const button = document.createElement('button');
+        const checkboxId = checkbox.id;
+        const checkboxControls = checkbox.getAttribute('aria-controls');
+        const checkboxClasses = Array.prototype.slice.call(checkbox.classList);
```

Comment:
I think `Array.from()` is more readable, but you're right - will be part of the follow-up PR.